src/hooks/useSearchTerms.ts

```diff
@@ -0,0 +1,173 @@
+import { FreeSearchFormObject, IOption } from '@/interfaces/common';
+import { ISearchForm } from '@/interfaces/formInterfaces';
+import {
+  clearSearchState,
+  getSearchResultsThunk,
+  initialSearchForm,
+  selectSubmittedSearchForm,
+  setLastSearchParams,
+  setSubmittedSearchForm,
+} from '@/reducers/searchSlice';
+import buildSearchParams from '@/utils/buildSearchParams';
+import { TFunction } from 'i18next';
+import { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppDispatch, useAppSelector } from './common';
+
+interface ISearchTerm {
+  value: string;
+  id: string;
+  type: string;
+}
+
+/**
+ * Build a search param with all the choices from the search form
+ *
+ * @param form (ISearchForm)
+ * @param translate (i18next translate function)
+ * @returns a list of ISearchTerm
+ */
+const getSearchTerms = (
+  form: ISearchForm,
+  translate: TFunction<'translation', undefined>,
+): Array<ISearchTerm> => {
+  const searchTerms = [];
+  for (const [key, value] of Object.entries(form)) {
+    switch (key) {
+      case 'masterClass':
+      case 'class':
+      case 'subClass':
+      case 'district':
+      case 'division':
+      case 'subDivision':
+        value.forEach((v: IOption) => {
+          const termLabel = translate(`searchTag.${key}`);
+          return searchTerms.push({
```

Comment:
return is redundant within forEach. it means it will break the loop.

Reply:
fixed